content/topics/sql-schema-design/index.md

```diff
@@ -0,0 +1,127 @@
+---
+title: SQL Database schema design
+ready: True
+---
+
+
+
+
+In this topic, we will discuss the schema design. Here is what Wikipedia says about "Database Schema":
+
+> The database schema of a database is its structure described in a formal language supported by the database management system. The term "schema" refers to the organization of data as a blueprint of how the database is constructed. [Wikipedia](https://en.wikipedia.org/wiki/Database_schema)
+
+To help us understand the design process we will design an Umuzi.org physical database as an example. We will use Entity Relationship Diagram  (ERD) as a method to design our schema, I have selected [dbdiagram](https://dbdiagram.io) as a  tool to help us achieve this objective, I find the tool painless and straightforward, it provides fast feedback and I get to download a physical database right after, feel free to explore alternatives on the internet or experiment along.
+
+This topic requires that you are a little bit familiar with [SQL](https://www.youtube.com/watch?v=27axs9dO7AE).
+
+## The brief
+
+
+As a developer working on a product you have been given a ticket that has the following requirements:
+>Design a Umuzi's department database schema to handle data related to it.
+
+The strategist on the team has added the following details to your ticket:
+> The database you are about to design should handle the following queries
+-  if the user has a recruit's full name he/she should be able to find which cohort the recruit belongs to.
+-  if the user has a cohort name he/she should be able to find recruit(s) in it.
+-  if the user has a staff member's email he/she should be able to retrieve recruits  he/she is servicing
+
+
+## The steps
+
+  Step 1:   idetify `entities` within a deparment
+  ---
+
+ Our "Umuzi" schema to include the following entities:
+
+ - Departments
+ - Cohorts
+ - Recruits
+ - staff
+
+![step 1 image](./img/step_1.png)
+
+
+  Step 2: Add more attributes
+  ---
+let establish all entities with their names and properties
+
+
+![step 2 image](./img/step_2.png)
+
+  Step 3: establish  relationships
+  ---
+
+  Each entity, attribute and relationship, should have appropriate names and that can be easily understood by the non-technical people as well. A relationship should connect entities.
+
+ We will consider the following relationships within our  database :
+
+- a one to one
+- a one to many
+- a many to many
+
+### CASE 1 : One  to one relationship
+
+Think on of a student who has joined a cohort,
```

Comment:
This is not a one-to-one relationship because **ONE** cohort can contain **MANY** recruits. 

If  ONE cohort could only contain ONE recruit then it would be a one-to-one. But this is not the case.

Please try one more time then if you can't finish this we need to ask someone else to. I believe Beard will be able to help here. He struggled as well.